cli/js/dispatch_json.ts

```diff
@@ -75,11 +75,17 @@ export async function sendAsync(
   const promiseId = nextPromiseId();
   args = Object.assign(args, { promiseId });
   const promise = util.createResolvable<Ok>();
-  promiseTable.set(promiseId, promise);
 
   const argsUi8 = encode(args);
-  const resUi8 = core.dispatch(opId, argsUi8, zeroCopy);
-  util.assert(resUi8 == null);
+  const buf = core.dispatch(opId, argsUi8, zeroCopy);
+  if (buf) {
+    // Sync result.
+    const res = decode(buf);
+    promise.resolve(res);
+  } else {
+    // Async result.
+    promiseTable.set(promiseId, promise);
+  }
```

Comment:
Is this necessary for the greedy?

Reply:
Yes, greedy poll happens in isolate, that means that `core.dispatch` can return sync result for async ops so we need to handle it appropriately in each dispatch mechanism

Reply:
Ah - of course! Thanks for the explanation.